src/System.IdentityModel.Tokens/ECDSASecurityKey.cs

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved
+// Apache License 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Security.Cryptography;
+using Microsoft.IdentityModel.Logging;
+
+namespace System.IdentityModel.Tokens
+{
+    public class ECDsaSecurityKey : AsymmetricSecurityKey
+    {
+        private byte[] _blob;
+        private CngKeyBlobFormat _blobFormat;
+        private CngKey _cngKey;
+
+        public ECDsaSecurityKey(byte[] blob, CngKeyBlobFormat blobFormat)
+        {
+            if (blob == null)
+            {
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10000, "ECDsaSecurityKey.blob"), typeof(ArgumentNullException), EventLevel.Verbose);
+            }
+
+            if (blobFormat == null)
+            {
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10000, "ECDsaSecurityKey.blobFormat"), typeof(ArgumentNullException), EventLevel.Verbose);
+            }
+
+            _cngKey = CngKey.Import(blob, blobFormat);
+            _blob = blob;
+            _blobFormat = blobFormat;
+        }
+
+        public override bool HasPrivateKey
+        {
+            get
+            {
+                return (_blobFormat.Format == CngKeyBlobFormat.EccPrivateBlob.Format || _blobFormat.Format == CngKeyBlobFormat.GenericPrivateBlob.Format);
```

Comment:
Nothing comes to mind. Might want to ask the WinCrypto guys on an internal email.